extract_endpoint/tests/test_endpoint.py

```diff
@@ -0,0 +1,165 @@
+import io
+import base64
+import pytest
+from flask import testing
+from azure.storage import blob
+from extract_endpoint import azure_utils
+from extract_endpoint import crypt_utils
+from extract_endpoint import endpoint
+
+
+endpoint.App.testing = True
+
+
+@pytest.fixture
+def azure_service() -> blob.BlockBlobService:
+    return blob.BlockBlobService(account_name=azure_utils.AZURE_EMULATOR_COORDS.account,
+                                 account_key=azure_utils.AZURE_EMULATOR_COORDS.key,
+                                 custom_domain=azure_utils.AZURE_EMULATOR_COORDS.domain)
+
+
+@pytest.fixture
+def test_client(azure_service: blob.BlockBlobService) -> testing.FlaskClient:
+    endpoint.App.config['AZURE_COORDINATES'] = azure_utils.AZURE_EMULATOR_COORDS
+    azure_service.create_container(azure_utils.AZURE_EMULATOR_COORDS.container)
+    yield endpoint.App.test_client()
+    azure_service.delete_container(azure_utils.AZURE_EMULATOR_COORDS.container)
+
+
+@pytest.fixture
+def sample_salt() -> str:
+    return 'sample salt'
+
+
+@pytest.fixture
+def sample_secret_key() -> str:
+    endpoint.App.config['SECRET_KEY'] = 'sample secret key'
+    return endpoint.App.config['SECRET_KEY']
```

Comment:
This fixture has side effects that it doesn't clean up, I'd save the current set `endpoint.App.config['SECRET_KEY']`, and reset it afterwards

Reply:
yes please. Fixtures/tests should not have any side effects beyond the fixture/test scope. `yield` is your friend here. If it were an environment variable, `monkeypatch.setenv` would do it.

Reply:
👍